src/libstd/sys/unix/rwlock.rs

```diff
@@ -9,16 +9,22 @@
 // except according to those terms.
 
 use libc;
-use cell::UnsafeCell;
+use cell::{Cell, UnsafeCell};
 
-pub struct RWLock { inner: UnsafeCell<libc::pthread_rwlock_t> }
+pub struct RWLock {
+    inner: UnsafeCell<libc::pthread_rwlock_t>,
+    write_locked: Cell<bool>,
```

Comment:
Doesn't `Cell` also allow concurrent access? Neither of them have the `Sync` marker, but that doesn't matter since that marker added to `RWLock` 4 lines later.